pandas/core/dtypes/missing.py

```diff
@@ -567,7 +567,22 @@ def _array_equivalent_object(left: np.ndarray, right: np.ndarray, strict_nan: bo
 
         return lib.array_equivalent_object(ensure_object(left), ensure_object(right))
 
-    for left_value, right_value in zip(left, right):
+    mask = isna(left) & isna(right)
+    try:
+        if not lib.array_equivalent_object(
+            ensure_object(left[~mask]),
```

Comment:
Is it worth skipping the masking if there are no nulls in left and right?

Reply:
good point. updated